src/bp-core/classes/class-bp-component-feature.php

```diff
@@ -0,0 +1,223 @@
+<?php
+/**
+ * BuddyPress Component's feature Class.
+ *
+ * @package BuddyPress
+ * @subpackage Core
+ * @since 15.0.0
+ */
+
+// Exit if accessed directly.
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * BuddyPress Component Feature API.
+ *
+ * @since 15.0.0
+ */
+class BP_Component_Feature extends BP_Component {
+
+	/**
+	 * Unique ID for the feature.
+	 *
+	 * @since 15.0.0
+	 *
+	 * @var string
+	 */
+	public $id = '';
+
+	/**
+	 * Unique Slug for the feature.
+	 *
+	 * @since 15.0.0
+	 *
+	 * @var string
+	 */
+	public $slug = '';
+
+	/**
+	 * Unique ID for the feature's component.
+	 *
+	 * @since 15.0.0
+	 *
+	 * @var string
+	 */
+	public $component_id = '';
+
+	/**
+	 * This is a component's feature.
+	 *
+	 * @since 15.0.0
+	 *
+	 * @var boolean
+	 */
+	public $is_feature = true;
+
+	/**
+	 * Disable the `BP_Component()` starting method.
```

Comment:
I think I understand that you're doing this to help differentiate features from components and essentially have two distinct setup routines. Maybe the note here should say explicitly that classes that extend BP_Component_Feature should use the init() method and classes that extend BP_Component should use start(). I'm not sold on this wording, but this section of the code confused me at first look, so I wonder if we can help code readers.